frontend/packages/console-app/src/components/cloud-shell/CloudShellTab.tsx

```diff
@@ -1,11 +1,17 @@
 import * as React from 'react';
 import { useTranslation } from 'react-i18next';
-import { InlineTechPreviewBadge } from '@console/shared';
+import { InlineTechPreviewBadge, useFlag } from '@console/shared';
+import { history } from '@console/internal/components/utils';
+import { FLAG_DEVWORKSPACE } from '../../consts';
 import CloudShellTerminal from './CloudShellTerminal';
 import './CloudShellTab.scss';
 
 const CloudShellTab: React.FC = () => {
   const { t } = useTranslation();
+  const devWorkspaceFlag = useFlag(FLAG_DEVWORKSPACE);
+
+  if (!devWorkspaceFlag) history.push('/');
```

Comment:
I think you should render a `<Redirect` component in this case instead of pushing to history. This will also replace the history instead of adding to the stack.

Reply:
IMO, this should be handled at the route level and not component level.

Reply:
Done.